src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -2179,32 +2140,77 @@ MulticopterPositionControl::control_position(float dt)
 void
 MulticopterPositionControl::generate_attitude_setpoint(float dt)
 {
-	/* reset yaw setpoint to current position if needed */
-	if (_reset_yaw_sp) {
-		_reset_yaw_sp = false;
-		_att_sp.yaw_body = _yaw;
-	}
 
-	/* do not move yaw while sitting on the ground */
-	else if (!_vehicle_land_detected.landed &&
-		 !(!_control_mode.flag_control_altitude_enabled && _manual.z < 0.1f)) {
+	/* calculate attitude setpoint from thrust vector */
```

Comment:
I would make two different methods one for flying manual mode (which should end up not to be in the position controller). And one for the thrust setpoint to attitude.